src/condvar.rs

```diff
@@ -29,6 +30,17 @@ impl WaitTimeoutResult {
     }
 }
 
+/// A type indicating how many times a thread was blocked during wait_while.
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+pub struct WaitWhileResult(u32);
+
+impl WaitWhileResult {
+    #[inline]
+    pub fn num_iters(self) -> u32 {
+        self.0
+    }
+}
```

Comment:
I don't think this is useful to track. I've never seen any use for this information. It also makes the API unnecessarily complex.

Reply:
I agree. I removed `WaitWhileResult` and updated the tests.